components/tidb_query_datatype/src/codec/collation/mod.rs

```diff
@@ -35,6 +37,24 @@ macro_rules! match_template_collator {
      }}
 }
 
+#[macro_export]
+macro_rules! match_template_charset {
+     ($t:tt, $($tail:tt)*) => {{
+         #[allow(unused_imports)]
+         use $crate::codec::collation::encoding::*;
+
+         match_template::match_template! {
+             $t = [
+                 UTF8 => EncodingUTF8,
+                 UTF8Mb4 => EncodingUTF8,
+                 Latin1 => EncodingLatin1,
```

Comment:
Why not this since Latin1 shares the same encoding behavior with UTF8:
```suggestion
                 Latin1 => EncodingUTF8,
```

Reply:
they are almost the same except return different error messages, see https://github.com/tikv/tikv/pull/11397/files#diff-765b6ea3a54d842fafd3e1113a1a296f915bd220a410618d532419e1bf2685eeR13